prereise/gather/griddata/hifld/data_process/topology.py

```diff
@@ -73,25 +116,53 @@ def connect_islands_with_minimum_cost(
     :param pandas.DataFrame subs: substation data frame indexed by ``ID`` with two
         columns ``LATITUDE`` and ``LONGITUDE`` representing the geographical
         coordination of each entry.
-    :param function cost_metric: a function defines the cost metric to calculate the
-        weight of lines, defaults to :py:func:`haversine`.
+    :param int island_size_lower_bound: smallest island the function should consider
+        exclusively, defaults to 0.
+    :param int island_size_upper_bound: largest island the function should consider
+        exclusively, defaults to None.
     :param dict state_neighbor: a dictionary defines the adjacency relationship among
         states, defaults to None and the constant dictionary ``abv_state_neighbor``
         defined in the ``const`` module is used.
+    :param func min_dist_method: the function used to calculate minimum distance
+        between two connected components, defaults to *naive* which uses
+        :py:func:`min_dist_of_2_conn_comp` or *kdtree" which uses
+        :py:func:`min_dist_of_2_conn_comp_kdtree`.
+    :param function cost_metric: a function defines the cost metric to calculate the
+        weight of lines, defaults to :py:func:`haversine`.
     :param bool memory_efficient: run the function in a memory efficient way or not,
         defaults to False.
+    :param dict kdtree_kwargs: keyword arguments to pass to
+        :py:func:`min_dist_of_2_conn_comp_kdtree` if specified by ``min_dist_method``.
     :return: (*tuple*) -- a pair of lists, the first one is a list of all potential
         line candidates among given connected components; the second one is a list
         of subsequence of the first entry, representing the chosen lines to form a
         minimum spanning tree. Each entry of both lists is a 3-tuple, index of the
         first connected component, index of the second connected component,
         a dictionary with keys containing ``start``, ``end`` and ``weight``, defines
         the ``from substation ID``, ``to substation ID`` and the weight of the line
-        calculated by the ``cost_metric``.
+        calculated by either ``cost_metric`` or KDTree based on ``kdtree_kwargs``.
+    :raises TypeError:
+        if ``island_size_lower_bound`` is not int, and/or
+        if ``island_size_upper_bound`` is not int, and/or
+        if ``kdtree_kwargs`` is specified but not a dict.
+    :raises ValueError:
+        if ``island_size_lower_bound`` is greater or equal to
+        ``island_size_upper_bound``, and/or
+        if ``min_dist_method`` is unknown.
 
         .. note:: the indexes of connected components are defined by the size,
             i.e. number of nodes, of the connected components in ascending order.
     """
+    if island_size_upper_bound is None:
+        island_size_upper_bound = len(subs) + 1
```

Comment:
Would `float("inf")` be clearer than `len(subs) + 1`?

Reply:
Actually, I thought about it. If we use `float("inf")` instead of `None`, it's a little confusing to give an `int` type variable a `float` default value, also, we need to take care of it during the input checking in l.160-l.161 of `topology.py`.

Reply:
Good point.